tests/ship_streamer_test.py

```diff
@@ -57,28 +63,98 @@
     cluster.killall(allInstances=killAll)
     cluster.cleanup()
     Print("Stand up cluster")
+
+
+    # ***   setup topogrophy   ***
+
+    # "bridge" shape connects defprocera through defproducerc (3 in node0) to each other and defproduceru (1 in node1)
+    # and the only connection between those 2 groups is through the bridge node
+
+    shipNodeNum = 1
     specificExtraNodeosArgs={}
-    # non-producing nodes are at the end of the cluster's nodes, so reserving the last one for state_history_plugin
-    shipNodeNum = totalNodes - 1
-    specificExtraNodeosArgs[shipNodeNum]="--plugin eosio::state_history_plugin --disable-replay-opts --trace-history --sync-fetch-span 200 --plugin eosio::net_api_plugin "
+    specificExtraNodeosArgs[shipNodeNum]="--plugin eosio::state_history_plugin --disable-replay-opts --trace-history --chain-state-history --plugin eosio::net_api_plugin "
+    # producer nodes will be mapped to 0 through totalProducerNodes-1, so the number totalProducerNodes will be the non-producing node
+    specificExtraNodeosArgs[totalProducerNodes]="--plugin eosio::test_control_api_plugin  "
 
-    if cluster.launch(pnodes=totalProducerNodes,
+    if cluster.launch(topo="bridge", pnodes=totalProducerNodes,
                       totalNodes=totalNodes, totalProducers=totalProducers,
                       useBiosBootFile=False, specificExtraNodeosArgs=specificExtraNodeosArgs) is False:
         Utils.cmdError("launcher")
         Utils.errorExit("Failed to stand up eos cluster.")
 
     # ***   identify each node (producers and non-producing node)   ***
 
-    shipNode = cluster.getNode(shipNodeNum)
-    prodNode = cluster.getNode(0)
-
     #verify nodes are in sync and advancing
     cluster.waitOnClusterSync(blockAdvancing=5)
     Print("Cluster in Sync")
 
+    prodNode = cluster.getNode(0)
+    prodNode0 = prodNode
+    prodNode1 = cluster.getNode(1)
+    nonProdNode = cluster.getNode(2)
+    shipNode = cluster.getNode(shipNodeNum)
+
+
+    accounts=cluster.createAccountKeys(6)
+    if accounts is None:
+        Utils.errorExit("FAILURE - create keys")
+
+    accounts[0].name="testeraaaaaa"
+    accounts[1].name="tester111111" # needed for voting
+    accounts[2].name="tester222222" # needed for voting
+    accounts[3].name="tester333333" # needed for voting
+    accounts[4].name="tester444444" # needed for voting
+    accounts[5].name="tester555555" # needed for voting
+
+    testWalletName="test"
+
+    Print("Creating wallet \"%s\"." % (testWalletName))
```

Comment:
Would prefer f-strings in new code. There are several throughout, but I'll only comment here on this one so if you choose not to update, there is only one comment to close.

Reply:
Changed all to use f-strings